protostar-rust/src/lib.rs

```diff
@@ -41,16 +76,25 @@ fn internal_run_tests(input_path: &str) -> anyhow::Result<()> {
     Ok(())
 }
 
-#[pyfunction]
-fn run_tests(input_path: String) -> PyResult<()> {
-    internal_run_tests(&input_path.to_owned())
-        .map_err(|e| PyErr::new::<RuntimeError, _>(format!("{:?}", e)))?;
+// #[pyfunction]
+// fn run_tests(input_path: String) -> PyResult<()> {
+//     internal_run_tests(&input_path.to_owned())
+//         .map_err(|e| PyErr::new::<RuntimeError, _>(format!("{:?}", e)))?;
+//
+//     Ok(())
+// }
+//
+// #[pymodule]
+// fn rust_test_runner_bindings(_py: Python, m: &PyModule) -> PyResult<()> {
+//     m.add_wrapped(wrap_pyfunction!(run_tests))?;
+//     Ok(())
+// }
 
-    Ok(())
-}
-
-#[pymodule]
-fn rust_test_runner_bindings(_py: Python, m: &PyModule) -> PyResult<()> {
-    m.add_wrapped(wrap_pyfunction!(run_tests))?;
-    Ok(())
+#[derive(Deserialize, Debug)]
+pub struct ProtostarTestConfig {
+    exit_first: Option<bool>,
+    ignore: Option<Vec<String>>,
+    json: Option<bool>,
+    last_failed: Option<bool>,
+    report_slowest_tests: Option<bool>,
```

Comment:
some of these fields could make use of default values I think?

Reply:
If not provided, these should be resolved to `None` I think. And we can then handle non values as `false` or whatever other default makes sense.

Reply:
That's true, but you can make your life easier by just doing:

```rust
#[serde(default)]
json: bool
```